### PR TITLE
SISRP-6547 - Campus Solutions splash page

### PR DESCRIPTION
**Pull request by winstonkim**
https://jira.berkeley.edu/browse/SISRP-6547

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,10 @@
+; editorconfig.org
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+indent_style = space
+indent_size = 2
+trim_trailing_whitespace = true
```
Comment (christianvuerings) [inline .editorconfig]: @winstonkim can we add an end of line here?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+language: node_js
+sudo: false
+node_js:
+# https://github.com/sass/node-sass/issues/490
+#  - "0.11"
+  - "0.10"
+before_script:
+  - gem install scss-lint
+  - npm install -g gulp
+script: gulp prod
```
Comment (christianvuerings) [inline .travis.yml]: @winstonkim looks like we don't have `gulp prod` can you look into this?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/ucberkeley/sis-splash-main"
+  },
+  "author": "Winston Kim",
+  "license": "ECL 2",
```
Comment (christianvuerings) [inline package.json]: @winstonkim "license": "ECL-2.0"

File: .scss-lint.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+exclude:
+  - 'node_modules/**'
+  - 'dist/**'
```
Comment (christianvuerings) [inline .scss-lint.yml:3]: @winstonkim can we remove it here as well?

File: README.md [hunk 5]
```diff
@@ -1,5 +1,10 @@
 # The SIS Splash Page
+This is the splash page for PeopleSoft consultants to log in to Campus Solutions. Consultants can access the log in portal through the splash page in addition to links to useful resources that a consultant might need.
```
Comment (christianvuerings) [inline README.md]: Also accessed by staff & faculty

File: .jscsrc [hunk 6]
```diff
@@ -0,0 +1,148 @@
+{
```
Comment (christianvuerings) [inline .jscsrc]: @winstonkim can we run this somewhere? `jscs`

File: gulpfile.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  }))
+  .pipe(gulp.dest('src/css'));
+});
+
+// Concatenate & Minify JS
+gulp.task('scripts', function() {
+  return gulp.src('js/*.js')
+  .pipe(concat('all.js'))
+  .pipe(gulp.dest('dist'))
+  .pipe(rename('all.min.js'))
```
Comment (christianvuerings) [inline gulpfile.js]: @winstonkim as discussed please remove all the JavaScript references.

File: out/index.html [hunk 8]
```diff
@@ -0,0 +1,233 @@
+<html>
+  <head>
+    <meta charset="utf-8" />
```
Comment (christianvuerings) [inline out/index.html]: @winstonkim alignment

File: out/index.html [hunk 9]
```diff
@@ -0,0 +1,233 @@
+<html>
```
Comment (christianvuerings) [inline out/index.html:2]: @winstonkim html5 doctype

File: src/index.html [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+    <!-- Set the viewport width to device width for mobile -->
+    <meta name="viewport" content="width=device-width, initial-scale=1.0, minimal-ui" />
+
+    <link rel="stylesheet" type="text/css" href="css/splash.css" inline>
+  </head>
+  <body>
+  	<div class="section section-0">
+  		<div class="container">
+  			<a title="To SIS Project Home" href="http://sisproject.berkeley.edu/">
+  				<svg class="logo" version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="-223 336.5 576 96.5" enable-background="new -223 336.5 576 96.5" xml:space="preserve">
```
Comment (christianvuerings) [inline src/index.html]: @winstonkim can we look at making this better.

File: src/index.html [hunk 11]
```diff
@@ -0,0 +1,233 @@
...
+                <li>
+                  <a href="https://calshare.berkeley.edu/sites/sait/sis/_layouts/15/WopiFrame.aspx?sourcedoc=/sites/sait/sis/Docs/ProjectEnvironmentsInfo.docx&action=default">SIS Environments</a>
+                </li>
+              </ul>
+            </div>
+            <a href="http://sis-staff.berkeley.edu/timeline-links">
+              <img alt="Highlevel Timeline" class="timeline-banner" src="http://sis-staff.berkeley.edu/sites/default/files/intranet-timeline-button.jpg">
+            </a>
+          </div>
+        </div>
```
Comment (christianvuerings) [inline src/index.html:72]: @winstonkim alignment

File: src/scss/splash.scss [hunk 12]
```diff
@@ -0,0 +1,267 @@
...
+    display: inline-block;
+    width: 49%;
+  }
+
+  ul {
+    list-style-type: none;
+    margin: 0;
+    padding: 0;
+  }
+}
```
Comment (christianvuerings) [inline src/scss/splash.scss:273]: @winstonkim Horizontal scrollbar?

File: src/scss/splash.scss [hunk 13]
```diff
@@ -0,0 +1,267 @@
...
+
+.iframe-container {
+  height: 0;
+  overflow: hidden;
+  padding-bottom: 50.25%;
+  padding-top: 30px;
+  position: relative;
+}
+
+.iframe-container iframe,
```
Comment (christianvuerings) [inline src/scss/splash.scss]: @winstonkim SASS rewrite?

File: src/scss/splash.scss [hunk 14]
```diff
@@ -0,0 +1,267 @@
...
+.timeline-banner {
+  border: solid 1px $light-gray;
+  width: 100%;
+}
+
+.container {
+  left: 50%;
+  max-width: 970px;
+  position: relative;
+  transform: translateX(-50%);
```
Comment (christianvuerings) [inline src/scss/splash.scss:193]: @winstonkim learned something new!

File: gulpfile.js (hunk 7 above)
Comment (winstonkim) [inline gulpfile.js]: @christianv sorry need some clarification. should we remove jshint and jscs files even when we are using gulpfile.js?

Comment (christianvuerings) [inline gulpfile.js]: @winstonkim yes, I would be fine with that.

**Comment by christianvuerings** [comment]
@winstonkim could you ping when when you've made the update?

**Comment by winstonkim** [comment]
@christianv made the update! even cleaned the svg up, but travis doesnt seem to be running?

File: out/index.html [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+ <div class="section section-0">
+  <div class="container">
+    <a title="To SIS Project Home" href="http://sisproject.berkeley.edu/">
+      <div class="logo"></div>
+    </a>
+  </div>
+</div>
+<div class="section section-1">
+  <div class="container">
+   <div class="square-container">
```
Comment (christianvuerings) [inline out/index.html]: @winstonkim indentation still looks a bit off, can you make sure we use the 2 spaces everywhere?

**Comment by christianvuerings** [comment]
@winstonkim one last comment and interesting thing about travis, can't seem to make it work even though it's enabled.

**Comment by christianvuerings** [comment]
@winstonkim travis fix should have been applied in #6 - Can you rebase (we now have a merge conflict) and leave your `travis.yml` changes out?

FYI - https://github.com/travis-ci/travis-core/pull/137

**Comment by christianvuerings** [comment]
@winstonkim looks like `gulp` is working now but the build is failing:

```
No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.
```

Can you have another look?

**Comment by winstonkim** [comment]
@christianv got it to pass!

File: gulpfile.js [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+// Exit for exit
+gulp.task('exit', function() {
+  process.exit(0);
+});
+
+// Default Task
+gulp.task('default', ['sass', 'watch', 'inlinesource']);
+
+// Test Task
+gulp.task('test', ['sass', 'watch', 'inlinesource', 'exit']);
```
Comment (christianvuerings) [inline gulpfile.js]: @winstonkim could you remove the `watch` from this task? Also, aren't those tasks run asynchronously?

Comment (winstonkim) [inline gulpfile.js]: @christianv made the changes, let me know if there are any other problems!

**Pull request merged**